engine/class_modules/warlock/sc_warlock.cpp

```diff
@@ -355,6 +355,17 @@ struct scouring_tithe_t : public warlock_spell_t
 
     LSD_alt_value = p->find_spell( 360953 )->effectN( 2 ).percent();
   }
+  
+  void impact( action_state_t* s ) override
+  {
+    warlock_spell_t::impact( s );
+
+    if ( td( s->target )->dots_scouring_tithe->remains() > p()->find_spell( 312321 )->duration() &&
```

Comment:
Please confirm whether or not this is triggering LSD on the initial cast with this code (unsure if the remains() check is happening after dot has been applied due this being placed after impact() call)

Also, find_spell should not be used during combat.

Reply:
So the cast was not triggered on initial cast, as the remaining time was equal to the spell duration. With that check it was only triggered when refreshed.

I tried a lot of things to check if the cast was running on the target at the end of the cast, and if yes, give LSD buff. But the check was always done after the end of cast so I could not check it that way.

Maybe you have an idea how it could have been made more cleanly...